standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1453,6 +1454,7 @@ public ColumnStatistics getTableStats(final String catName, final String dbName,
     };
     List<Object[]> list = Batchable.runBatched(batchSize, colNames, b);
     if (list.isEmpty()) {
+      b.closeAllQueries();
       return null;
     }
```

Comment:
Same here....

```java
    List<Object[]> list = Batchable.runBatched(batchSize, colNames, b);
    final ColumnStatistics result;
    if (list.isEmpty()) {
        result = null;
    } else {
       ColumnStatisticsDesc csd = new ColumnStatisticsDesc(true, dbName, tableName);
       csd.setCatName(catName);
       result = makeColumnStats(list, csd, 0, engine);
    }
    b.closeAllQueries();
    return result;
  }
```

Reply:
Appreciate that, done. Thank you very much for the review!